lib/endpoints/metrics.js

```diff
@@ -17,12 +17,50 @@ var mod_restify = require('restify');
 
 var lib_errors = require('../errors');
 
+function fetchAndSendMetrics(req, res, req2, next) {
+    req2.on('result', function _fetchMetrics(err2, res2) {
+        if (err2) {
+            req.log.error(err2, 'Error fetching metrics');
+            next(new lib_errors.InternalServerError());
+            return;
+        }
+
+        var encoding;
+        if ((encoding = res2.header('Content-Encoding'))) {
+            res.header('Content-Encoding', encoding);
+        }
+
+        res.header('content-type', 'text/plain');
+
+        var data = [];
+        res2.on('data', function _onData(chunk) {
+            data.push(chunk);
+        }).on('end', function _onEnd() {
+            var buffer = Buffer.concat(data);
+            res.end(buffer);
+            next();
+        });
+    });
+}
+
 function apiGetMetrics(req, res, next) {
     var serverCacheItem, admin_ip, cmon_agent_url, cmon_agent_path, client;
 
     var target_uuid = req.header('HOST').split('.')[0];
     var cache = req.app.cache;
+    var config = req.app.config;
+
     mod_assert.object(cache, 'cache object');
+    mod_assert.object(config, 'config object');
+
+    var encoding;
+    var headers = {};
+    var server_opts = config.server_opts;
```

Comment:
Are we missing an update to the [SAPI template](https://github.com/joyent/triton-cmon/blob/master/sapi_manifests/cmon/template) here? I'm not sure where server_opts is attached to the config objected currently.

Reply:
I updated the template [here](https://github.com/joyent/triton-cmon/pull/14/commits/ec25498866c1e995ef57c6efe163d247bd34304f#diff-51681ee929cd04032b5d999c35190806R25-R27). Am I missing something?

Reply:
Ah somehow I missed that file being modified, sorry about that.